src/huggingface_hub/utils/_auth.py

```diff
@@ -121,6 +125,88 @@ def _get_token_from_file() -> Optional[str]:
         return None
 
 
+def get_profiles() -> Dict[str, str]:
+    """
+    Returns the parsed INI file containing the auth profiles.
+    The file is located at `HF_PROFILES_PATH`, defaulting to `~/.cache/huggingface/profiles`.
+    If the file does not exist,
+
+    Returns: `Dict[str, str]`
+        Key is the profile name and value is the token.
+
+    Raises:
+        FileNotFoundError: If the profiles file does not exist.
+    """
+    profiles_path = Path(constants.HF_PROFILES_PATH)
+    if not profiles_path.exists():
+        profiles = {}
+    config = configparser.ConfigParser()
+    try:
+        config.read(profiles_path)
+        profiles = {profile: config.get(profile, "hf_token") for profile in config.sections()}
+    except configparser.Error as e:
+        logger.error(f"Error parsing profiles file: {e}")
+        profiles = {}
+    return profiles
+
+
+def _save_profiles(profiles: Dict[str, str]) -> None:
+    """
+    Saves the given configuration to the profiles file.
+
+    Args:
+        profiles (`Dict[str, str]`):
+            The profiles to save. Key is the profile name and value is the token.
+    """
+    profiles_path = Path(constants.HF_PROFILES_PATH)
+
+    # Write the profiles into an INI file
+    config = configparser.ConfigParser()
+    for profile_name, token in profiles.items():
+        config.add_section(profile_name)
+        config.set(profile_name, "hf_token", token)
```

Comment:
```suggestion
    for profile in sorted(profiles.keys()):
        config.add_section(profile)
        config.set(profile, "hf_token", profiles[profile])
```

(nit) for better UX, I would sort the profiles in the file. This way they'll be displayed sorted in the UI as well. Usually makes it easier to find the correct one